### PR TITLE
Fix godunov gpu

### PR DESCRIPTION
**Pull request by cgilet**
Godunov on GPU is not quite working yet in IAMR, but at least this fix can be merged so PeleLM can have it.

### CODE REVIEW

File: Source/Godunov/iamr_godunov.cpp [hunk 1]
```diff
@@ -57,13 +57,16 @@ Godunov::ComputeAofs ( MultiFab& aofs, const int aofs_comp, const int ncomp,
 
         if (not known_edgestate)
         {
+	    auto d_bc  = convertToDeviceVector(bcs);
+	    BCRec const* d_bc_ptr = d_bc.data();
+
```
Comment (drummerdoc) [inline Source/Godunov/iamr_godunov.cpp:63]: Does it make sense to do this once and save the results as class data?

**Review by drummerdoc — APPROVED** [review]

File: Source/Godunov/iamr_godunov.cpp (hunk 1 above)
Comment (cgilet) [reply Source/Godunov/iamr_godunov.cpp:63]: Darn you and your valid point.

**Review by drummerdoc — COMMENTED** [review]
Wahoo, "several"!  :)

**Comment by drummerdoc** [comment]
Which case runs to completion?

**Comment by cgilet** [comment]
All of them. There might be a problem with the Rayleigh-Taylor regression test with USE_CUDA=TRUE (it uses ppm where most tests use plm), but I haven't looked into this yet. Also note that the create_umac_grown fix changes some answers, even with USE_CUDA=FALSE.

**Comment by cgilet** [comment]
I should add that I haven't tested PLM yet. Want to get that done that before merging.

**Comment by cgilet** [comment]
@drummerdoc I think we should go ahead and merge this and the accompanying PLM PR#113 since they contain bug fixes that can be relevant to CPU only runs. I've checked PeleLM results on CPU for 2d, 3d, EB and non-EB, and these PRs don't introduce changes above what's expected given problem tolerances. I've confirmed that IAMR changes to CPU run are due to the create_umac_grown() bug fixed here. 
On GPU, IAMR plm agrees with CPU only results. IAMR ppm still needs debugging. PeleLM also needs further debugging.

**Pull request merged**